app.rb

```diff
@@ -0,0 +1,136 @@
+require_relative './person'
+require_relative './student'
+require_relative './teacher'
+require_relative './classroom'
+require_relative './book'
+require_relative './rental'
+
+class App
+  attr_accessor :books, :people, :rentals
+
+  def initialize
+    @books = []
+    @people = []
+    @rentals = []
+  end
+
+  def entry_point
+    puts 'Welcome to School Library App!'
+
+    loop do
+      app_options
+      app_choice = gets.chomp.to_i
+      break if app_choice == 7
+
+      options(app_choice)
+    end
+  end
+
+  def app_options
+    puts ''
+    puts 'Please choose an option by entering a number:'
+    options = [
+      '1 - List all books',
+      '2 - List all people',
+      '3 - Create a person',
+      '4 - Create a book',
+      '5 - Create a rental',
+      '6 - List all rentals for a given person id',
+      '7 - Exit'
+    ]
+    puts options
+  end
+
+  def list_books
+    if @books.empty?
+      puts 'No books found'
+    else
+      @books.each_with_index { |book, index| puts "#{index}) Title: #{book.title}, Author: #{book.author}" }
+    end
+  end
+
+  def list_people
+    if @people.empty?
+      puts 'No people found'
+    else
+      @people.each_with_index do |person, index|
+        puts "#{index}) Name: #{person.name}, Age: #{person.age}, ID: #{person.id}"
+      end
+    end
+  end
+
+  def create_book
+    print 'Title: '
+    title = gets.chomp
+    print 'Author: '
+    author = gets.chomp
+    book = Book.new(title, author)
+    @books << book
+    puts 'Book created successfully'
+  end
+
+  def create_person
+    print 'Do you want to create a student (1) or a teacher (2)? [Input the number]: '
+    num_input = gets.chomp.to_i
+    case num_input
+    when 1
+      create_student
+    when 2
+      create_teacher
+    else
+      puts 'Invalid Entry'
+    end
+
+    puts 'Person Created Successfully'
+  end
+
+  def create_student
+    print 'Age: '
+    age = gets.chomp
+
+    print 'Name: '
+    name = gets.chomp.capitalize
+
+    print 'Has parent\'s permission? [Y/N]: '
+    parent_permission = gets.chomp.downcase == 'y'
+
+    @people << Student.new(@classroom, age, name, parent_permission: parent_permission)
+  end
+
+  def create_teacher
+    print 'Age: '
+    age = gets.chomp
+
+    print 'Name: '
+    name = gets.chomp.capitalize
+
+    print 'Specialization: '
+    specialization = gets.chomp.downcase
+
+    @people << Teacher.new(specialization, age, name)
+  end
+
+  def create_rental
+    puts 'Select a book from the following list by number'
+    list_books
+    selected_book = gets.chomp.to_i
+    puts 'Select a person from the following list by number'
+    list_people
+    selected_person = gets.chomp.to_i
+    print 'Date (DD/MM/YYYY): '
+    date = gets.chomp
+    book = @books[selected_book]
+    person = @people[selected_person]
+    rental_item = Rental.new(date, book, person)
```

Comment:
- [ ] Also when I try to create `rental` the console app failed, same error `wrong number of arguments`, kindly fix that as well 🙃 😉 

<img width="1168" alt="Screen Shot 2023-01-07 at 14 20 47" src="https://user-images.githubusercontent.com/74501523/211147609-4636ee8d-27d1-4184-bcb3-cf384fc35725.png">